### ISSUE TITLE
Pop doesnt seem to work

### ISSUE THREAD

**Issue by mlakewood** [salesforce/runway-compiler#5]
Hi all,

runway-compiler: commit 135ffcb,
nodejs: v6.2.1 from docker container node:6.2.1-wheezy

Im really excited about what runway could do for designing distributed system. An issue im running into however is the use of the pop() function. Here is an example repl session.

```
> type Number : 0..9;
> var os : OrderedSet<Number>[1..3];
> push(os, 2)
> push(os, 4)
> os
{1: 2, 2: 4}
> var e1 : Number = pop(os);
> e1
0
> os
{1: 2}
> var e2 : Number = pop(os);
> e2
4
> os
{}
> var e3 : Number = pop(os);
ModelingBoundsError: Cannot pop from empty {}
```

If I understand the documentation correctly pop() should return the last element of a collection. And in this case Im not exactly sure what is happening.

The first pop() return value is 0 which isnt a value inserted or an index available, and then we pop() again and we get the value that we should have got last time. If we pop again on the empty list it doesnt seem to work.

If im using pop() wrong (which is entirely likely) then let me know, as I couldnt see it being used in other places, only remove.

Thanks again for all the hard work, I really like the idea.

Mark

**Comment by ongardie-sfdc** [comment]
Thanks for your interest and for the really clear bug report, @mlakewood. I took your example and created a unit test from it, and I think I've now fixed this issue in cf74446. I also deployed that to https://runway.systems. The collections aren't very well-tested yet, so it's possible there's more trouble ahead -- if so, please keep the bug reports coming.

Cheers,
Diego

**Comment by mlakewood** [comment]
Thanks for the quick fix. I'll keep pushing forward :)
